### PR TITLE
Add functionality to make the scheduler a http service and also to ge…

### PR DESCRIPTION
**Pull request by reinhartfimqs**
…nerate a agp file

### CODE REVIEW

File: cmd/scheduler.go [hunk 1]
```diff
@@ -166,7 +169,60 @@ func loadConfig() {
...
+		return
+	}
+
+	// Change the importer command and then start the import job immediately
+	var cmds []*scheduler.Command
+	newCommand.StartTime = time.Now() // Set the importer start time to now so that it will run immediately
+	cmds = append(cmds, newCommand)
+
+	next := scheduler.NextRunnable(cmds, time.Now())
+	if next != nil {
```
Comment (LeonLeibbrandt) [inline cmd/scheduler.go]: Short circuit:
next := scheduler.NextRunnable([]*scheduler.Command{newCommand}, time.Now())

File: cmd/scheduler.go [hunk 2]
```diff
@@ -166,7 +169,60 @@ func loadConfig() {
...
+// Currently this is build specifically to generate a AGP file
+// in a specific location and then kick of a imqstool importer job.
+func handleHttpFunction(w http.ResponseWriter, r *http.Request) {
+	// Retrieve the query from the url in the request
+	parsedUrl, err := url.Parse(r.RequestURI)
+	if err != nil {
+		http.Error(w, fmt.Sprint(err), http.StatusInternalServerError)
+	}
+	fileName := parsedUrl.Query().Get("filename")
+
```
Comment (LeonLeibbrandt) [inline cmd/scheduler.go]: filename := r.FormValue("filename")

File: cmd/scheduler.go [hunk 3]
```diff
@@ -166,7 +169,60 @@ func loadConfig() {
...
+	}
+
+	// Create the AGP file required for the shape file import to work
+	if scheduler.CreateAGP(config.ImportPath, fileName) != true {
+		http.Error(w, "Could not create AGP file", http.StatusInternalServerError)
+		return
+	}
+
+	// Look for the Importer command in the list of commands
+	var newCommand *scheduler.Command
```
Comment (LeonLeibbrandt) [inline cmd/scheduler.go]: Cannot use pointer here as it will change the one in the original list as well.

**Comment by bmharper** [comment]
I don't think it's OK to generate an AGP file inside this service. That should be done from the ImqsCpp service. The ImqsCpp service generates the .agp file and places it alongside any other data that is going to be imported.
The reason we're making the scheduler an HTTP service, is so that the ImqsCpp service can communicate to the scheduler, saying "There is a new import job waiting to be run. Kick off the 'Imqs Importer' job immediately".

**Comment by bmharper** [comment]
Also, the title line of your commit is too long. If GitHub is generating an ellipsis (...) for your commit line, then you need to shorten it.
I recommend using the command-line git commit with vim, because it highlights this for you.

**Pull request closed without merging**